### ISSUE TITLE
The “i” key doesn’t work within Steam

### ISSUE THREAD

**Issue by deterenkelt** [ValveSoftware/steam-for-linux#5656]
#### Your system information

* Steam client version: Aug 8 2018 at 21:32:49 
* Distribution: Gentoo x86_64 
* Opted into Steam client beta?: No
* Have you checked for system updates?: Yes
 ­

#### Please describe your issue in as much detail as possible:

The <kbd>i</kbd> key should work, but it does not. 

For example, when I write to a friend in Steam chat, <kbd>i</kbd> and the next key, that I type after it, are “gulped”. They take no space on the screen, so “continuation” becomes “contuatn”. 

The similar thing happens even when I run a game. For example, in The Long Dark the inventory was always bound to the <kbd>i</kbd> key, and this key stopped working after a Steam update, that came a couple of weeks ago. I cannot reassign the inventory to <kbd>i</kbd> – the game engine behaves like it doesn’t see it. I can reassign it to <kbd>o</kbd>, – however, if I press <kbd>o</kbd> immediately after <kbd>i</kbd>, it won’t work. Left4Dead2 has the same issues.

This isn’t a problem with my keyboard, as I type this message without any troubles (Steam is running).

I’ve restarted and reinstalled Steam – it didn’t help.

 ­

#### Steps for reproducing this issue:

1. Run Steam
2. Try to chat to a friend or run a game
3. See that the <kbd>i</kbd> key isn’t working, and additionally gulps whatever key you press next.

**Comment by kisak-valve** [comment]
Hello @deterenkelt, in a quick test my gentoo test box is not affected. What model keyboard do you have?

**Comment by deterenkelt** [comment]
It’s a Samsung KB-0325. A generic keyboard with pc-104 layout.

**Comment by deterenkelt** [comment]
I’ve just tried Genius LuxeMate i202 (this one is USB, while Samsung is PS/2) – the issue is still there.

**Comment by deterenkelt** [comment]
I’ve just noticed that the same gulping happens within Vivaldi browser, which is based on chromium.

**Comment by deterenkelt** [comment]
I figured, that the troublemaker was this line in the `~/.XCompose` file:

    <i> <p> : "·" U00B7

A long time ago I attempted to set it up. I’ve dropped the idea, but the file remained in my `$HOME`. Due to `GTK_IM_MODIFIER` and `QT_IM_MODIFIER` being set to `scim`, the issue didn’t affect any other program except those that are based on Google’s v8.

I’ve discovered the root of the problem, when I removed scim from my system, and immediately got the same bug with the `i` key in urxvt. Which was strange, so I disabled things one by one, and when I couldn’t figure out even then, I’ve run `xev` to see what happens, when I press the `i` key. 

And there was something: each time on the KeyPress event for `i` and for the next key pressed after to it, XFilterEvent value returned true. Search on XFilterEvent has lead me to XCompose.
